src/services/keycloak.js

```diff
@@ -6,18 +6,19 @@ export default class KeycloakClient {
     constructor( config ) {
         this.host = config.host
         this.clientId = config.clientId
-        this.clientSecret = config.clientSecret
+        this.clientSecret = config.clientSecret,
+        this.extraOptions = config.extraOptions || {}
     }
 
     async ping() {
-        return rp( {
+        return rp( Object.assign({
             method: 'GET',
             uri: `${this.host}`
-        } )
+        }, this.extraOptions) )
     }
 
     async authenticate( username, password ) {
-        return rp( {
+        return rp( Object.assign({
```

Comment:
@Magnitus- although I understand the intent... I think we need to make some changes to make it resilient:

![image](https://user-images.githubusercontent.com/3919995/78025508-28494d00-7328-11ea-87df-18e657bc7528.png)

Reply:
I think that can be achieved by reversing the order of the arguments in Object.assign, I'll do that.

Reply:
Done.